cdap-data-fabric/src/main/java/co/cask/cdap/data2/metadata/lineage/field/FieldLineageInfo.java

```diff
@@ -382,41 +389,188 @@ private void computeIncomingSummaryHelper(EndPointField destination, Operation c
+    return visitedOperations;
+  }
+
+  /**
+   * Recursively traverse the graph to calculate the incoming operation.
+   *
+   * @param currentOperation the current operation from which the graph needs to explored
+   * @param visitedOperations all the operations visited so far
+   */
+  private void getIncomingOperationsForFieldHelper(Operation currentOperation, Set<Operation> visitedOperations) {
+    if (!visitedOperations.add(currentOperation)) {
+      return;
+    }
+
+    // reached the end of backward traversal
+    if (currentOperation.getType() == OperationType.READ) {
+      return;
+    }
+
+    // for transform we traverse backward in graph further through the inputs of the transform
+    if (currentOperation.getType() == OperationType.TRANSFORM) {
+      TransformOperation transform = (TransformOperation) currentOperation;
+      for (InputField field : transform.getInputs()) {
+        getIncomingOperationsForFieldHelper(operationsMap.get(field.getOrigin()), visitedOperations);
+      }
+    }
+  }
+
+  /**
+   * <p>Get the subset of operations that used the specified field</p>
+   *
+   * <p>For example if the operation are as follow</p>
+   * <pre>
+   * pRead: personFile -> (offset, body)
+   * parse: body -> (id, name, address)
+   * cRead: codeFile -> id
+   * codeGen: (parse.id, cRead.id) -> id
+   * sWrite: (codeGen.id, parse.name, parse.address) -> secureStore
+   * iWrite: (parse.id, parse.name, parse.address) -> insecureStore
+   * </pre>
+   *
+   * <p>The sourceField is 'id' of codeFile then the returned operations set will contain cRead, codeGen, sWrite.</p>
+   * <p>If the sourceField is 'body' of personFile then the returned set of operations will contain pRead, parse,
+   * codeGen, sWrite, iWrite.</p>
+   *
+   * @param sourceField the {@link EndPointField} whose outgoing operations needs to be found
+   * @return {@link Set} of {@link Operation} which are outgoing from the given sourceField
+   */
+  Set<Operation> getOutgoingOperationsForField(EndPointField sourceField) {
+    if (readOperations == null) {
+      computeAndValidateFieldLineageInfo(this.operations);
+    }
+
+    Set<Operation> visitedOperations = new HashSet<>();
+    for (ReadOperation readOperation : readOperations) {
+      if (readOperation.getSource().equals(sourceField.getEndPoint())) {
+        // if the read operation is for the dataset to which the sourceField belong
+        List<String> outputs = readOperation.getOutputs();
+        for (String output : outputs) {
+          // if the current read operation did read the sourceField for which outgoing operation is requested we
+          // consider it else not.
+          if (output.equals(sourceField.getField())) {
+            visitedOperations.add(readOperation);
+            for (Operation outgoingOperation : operationOutgoingConnections.get(readOperation.getName())) {
+              // check if the outgoing operation does use the sourceField else that even though is an outgoing
```

Comment:
I think it would be more clear to do say something like:

```
Check that the source field is an input field for the outgoing operation.
Consider the example in the method javadoc with:
  sourceField = personFile.offset
  readOperation = pRead: personFile -> (offset, body)
  outgoingOperation = parse: body -> (id, name, address)

In this scenario, 'offset' is not an input to the outgoingOperation so we do not need to go further down the graph.
If the sourceField was personFile.body, we would need to continue as 'body' is an input to the outgoingOperation.
```